src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -1032,6 +1032,10 @@ impl Step for PlainSourceTarball {
                 .arg(builder.src.join("./compiler/rustc_codegen_gcc/Cargo.toml"))
                 .arg("--sync")
                 .arg(builder.src.join("./src/bootstrap/Cargo.toml"))
+                .arg("--sync")
```

Comment:
I'm not sure if we should do this unconditionally, since `opt-dist` is only used in two CI dist jobs, most of them don't use it at all :thinking:

Reply:
> since opt-dist is only used in two CI dist jobs, most of them don't use it at all

That's true within the Rust project, but my main goal here is to make `opt-dist` usable for packagers / end users who need to bootstrap Rust themselves, but still want to benefit from PGO and BOLT.

Reply:
> I'm not sure if we should do this unconditionally

I wish we had a "src tarball for developers", which would not include any submodules for example, and a "src tarball for distributing" which would be "complete", but that would be a pretty large yak to shave.

Reply:
Which "developers" do you think are using it?

Reply:
For future reference, my argument boils down to:

**premises**

* the point of the `rustc-dev-src` dist tarball is to allow people to bootstrap Rust
* most people bootstrapping Rust want a production-grade toolchain as a result
* the vanilla bootstrap is not sufficient to bootstrap a production-grade rust-toolchain
* it is undesirable for users to be forced to reinvent the wheel and roll their own complex PGO/BOLT bootstrapping pipeline

**therefore**
* it should be possible to bootstrap a production-grade Rust toolchain from the `rustc-dev-src` tarball using `opt-dist`